parsers/lib/antlr4/LangJavaParser.g4

```diff
@@ -373,13 +376,21 @@ literal
 // ANNOTATIONS
 
 annotation
-    :   '@' annotationName ( '(' ( elementValuePairs | elementValue )? ')' )?
+    :   '@' annotationName annotationArgs?
+    ;
+
+annotationArgs :   '(' annotationArgsElems? ')' ;
+
+annotationArgsElems
+    : elementValuePairs
+    | elementValue
     ;
 
 annotationName : qualifiedName ;
 
 elementValuePairs
-    :   elementValuePair (',' elementValuePair)*
+    :   elementValuePair
+    |   elementValuePairs ',' elementValuePair
```

Comment:
Could this cause the effect in which the orders of the elements are different? If yes, would this cause problems? This question is also for the other parts that changed

Reply:
I don't think the order of the elements changed, they are still parsed left to right.

But the storage did: previously antlrs parsetree would contain a flat array of elementValuePairs I think (interspersed with "," nodes), and now each parse tree node contains one subtree containing many element value pairs, one "," node, and one elementvaluepair. So I guess it's a bit easier to use now, though a bit more verbose. 

If there was any existing code that used this part of the grammar I probably would've broke it, but this part of Java is not supported I think so this change is okay I'd say.

Reply:
Sounds good